### PR TITLE
feat(Core/Config): allow initial (free) amount of Guild Bank tabs

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
## Changes Proposed:
-  Add a new config in `worldserver.conf` allowing guilds to have an initial (free) amount of guild tabs

## Tests Performed:

Tested in game with a new guild after each `.reload conf`:

- 0 free tabs (default behaviour)
- 1 free tab
- 3 free tabs
- 6 free tabs (see screenshot below)

image

## How to Test the Changes:

1) Add/change the `Guild.BankInitialTabs` config in `worldserver.conf` (allowed values 0-6)
2) `.reload conf`
3) Create a new guild `.guild create "Guild Name"`
4) See if the Guild Bank has the correct amount of initial tabs

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Kitzunu** [comment]
I suppose it is an issue with auto :)

```
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/game/Guilds/Guild.cpp:1207:24: fatal error: comparison of integers of different signs: 'int' and 'uint32' (aka 'unsigned int') [-Wsign-compare]
    for (auto i = 0; i < sWorld->getIntConfig(CONFIG_GUILD_BANK_INITIAL_TABS); i++)
                     ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Review by Helias — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
> I suppose it is an issue with auto :)

@kitzunu it's because `auto i = 0` defaults to `int` while the `sWorld->getIntConfig` function returns `uint32`, thus the error:

```
comparison of integers of different signs: 'int' and 'uint32' (aka 'unsigned int')
```

solved by using `short` for both instead, as the values must always be between 0 and 6 so no need to use a bigger type.

**Comment by pklloveyou** [comment]
Tested

File: src/server/game/Guilds/Guild.cpp [hunk 1]
```diff
@@ -1203,6 +1203,12 @@ bool Guild::Create(Player* pLeader, std::string const& name)
     _CreateDefaultGuildRanks(pLeaderSession->GetSessionDbLocaleIndex()); // Create default ranks
     bool ret = AddMember(m_leaderGuid, GR_GUILDMASTER);                  // Add guildmaster
 
+
```
Comment (Kitzunu) [inline src/server/game/Guilds/Guild.cpp:1206]: I aint gonna let you get away with adding too many blank lines.

Remove it :D

Comment (FrancescoBorzi) [reply src/server/game/Guilds/Guild.cpp]: hehe, removed!

**Comment by BarbzYHOOL** [comment]
very good idea

**Comment by FrancescoBorzi** [comment]
@BarbzYHOOL why not, feel free to open an issue with it so we don't forget. Could be also a [Good first issue].

**Pull request merged**